src/io/image_aug_default.cc

```diff
@@ -136,6 +139,8 @@ struct DefaultImageAugmentParam : public dmlc::Parameter<DefaultImageAugmentPara
     DMLC_DECLARE_FIELD(pad).set_default(0)
         .describe("Change size from ``[width, height]`` into "
                   "``[pad + width + pad, pad + height + pad]`` by padding pixes");
+    DMLC_DECLARE_FIELD(seed_aug).set_default(-1)
```

Comment:
better use dmlc::optional<int>() and check has_value() rather than compare with -1. 
Literally we can use any number as seed.